plugin/src/main/java/org/owasp/benchmarkutils/score/parsers/SnykReader.java

```diff
@@ -0,0 +1,116 @@
+                    put("DOMXSS", CweNumber.XSS);
+                    put("XSS", CweNumber.XSS);
+                    put("InsecureCipherNoIntegrity", CweNumber.WEAK_CRYPTO_ALGO);
+                    put("InsecureDefaultAesCipher", CweNumber.WEAK_CRYPTO_ALGO);
+                    put("HttpResponseSplitting", CweNumber.HTTP_RESPONSE_SPLITTING);
+                    put("InsecureSecret", CweNumber.WEAK_RANDOM);
+                    put("LdapInjection", CweNumber.LDAP_INJECTION);
+                    put("InsecureCipher", CweNumber.WEAK_CRYPTO_ALGO);
+                    put("InsecureHash", CweNumber.WEAK_HASH_ALGO);
+                }
+            };
+
+    @Override
+    public boolean canRead(ResultFile resultFile) {
+        return resultFile.isJson() && isSnyk(resultFile);
+    }
+
+    @Override
+    public TestSuiteResults parse(ResultFile resultFile) throws Exception {
+        TestSuiteResults tr = new TestSuiteResults("Snyk", true, TestSuiteResults.ToolType.SAST);
+
+        JSONArray results =
+                resultFile.json().getJSONArray("runs").getJSONObject(0).getJSONArray("results");
+
+        for (int result = 0; result < results.length(); result++) {
+            TestCaseResult tcr = parseSnykFindings(results.getJSONObject(result));
+            if (tcr != null) {
+                tr.put(tcr);
+            }
+        }
+        return tr;
+    }
+
+    private TestCaseResult parseSnykFindings(JSONObject result) {
+        try {
+            String className =
+                    result.getJSONArray("locations")
+                            .getJSONObject(0)
+                            .getJSONObject("physicalLocation")
+                            .getJSONObject("artifactLocation")
+                            .getString("uri");
+            className = (className.substring(className.lastIndexOf('/') + 1)).split("\\.")[0];
+            if (className.startsWith(BenchmarkScore.TESTCASENAME)) {
+
+                TestCaseResult tcr = new TestCaseResult();
+
+                String ruleId = result.getString("ruleId");
+                ruleId = (ruleId.substring(ruleId.lastIndexOf('/') + 1)).split("\\.")[0];
+
+                int cwe = snykCweMap.getOrDefault(ruleId, INVALID_RULE_ID);
+
+                if (cwe == INVALID_RULE_ID) {
+                    System.out.println("CWE # not parseable from: " + ruleId);
+                    return tcr;
+                }
+
+                String evidence = result.getJSONObject("message").getString("text");
+
+                tcr.setCWE(cwe);
+                tcr.setCategory(ruleId);
+                tcr.setEvidence(evidence);
+                tcr.setConfidence(0);
+                tcr.setNumber(testNumber(className));
+
+                return tcr;
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+
+        return null;
+    }
+
+    private Boolean isSnyk(ResultFile resultFile) {
+
+        return resultFile.json().has("runs")
```

Comment:
One additional thing that came up to my mind this morning. On your path to the string `SnykCode` you use `has` method to check some, but not all parts of the structure. This still might throw an JSON exception. I got around this by using a try-catch and not check for the path at all (see `HorusecReader.canRead()`). Maybe you can just use a try-catch instead?